joint_trajectory_controller/src/trajectory.cpp

```diff
@@ -270,6 +300,62 @@ void Trajectory::interpolate_between_points(
                                 t[2] * 12.0 * coefficients[4] + t[3] * 20.0 * coefficients[5];
     }
   }
+  // Apply velocity, acceleration, and jerk limits with Ruckig
+  if (period.seconds() > 0)
+  {
+    ruckig::Ruckig<ruckig::DynamicDOFs> smoother(dim, period.seconds());
```

Comment:
It might seem strange to declare a new smoothing object with every cycle, but `period` could change. So I think it has to be done this way.